source/common/common/utility.cc

```diff
@@ -517,6 +517,33 @@ std::string StringUtil::removeCharacters(const absl::string_view& str,
   return absl::StrJoin(pieces, "");
 }
 
+absl::optional<uint64_t> StringUtil::readAndRemoveLeadingDigits(absl::string_view& str) {
```

Comment:
Please consider using the shorter and simpler approach of:

// If s is non-null and begins with a decimal number ([0-9]+), removes it from
// the input and returns a SystemTime::duration representing that many seconds.
// If s is null or doesn't begin with digits, returns
// SystemTime::duration::zero(). If parsing overflows, returns
// SystemTime::duration::max().
SystemTime::duration HttpCacheUtils::eatLeadingDuration(absl::string_view& s) {
  const absl::string_view::iterator digits_end = absl::c_find_if_not(s, &absl::ascii_isdigit);
  const size_t digits_length = digits_end - s.begin();
  if (digits_length == 0) {
    return SystemTime::duration::zero();
  }
  const absl::string_view digits(s.data(), digits_length);
  s.remove_prefix(digits_length);
  uint64_t num;
  return absl::SimpleAtoi(digits, &num) ? std::chrono::seconds(num) : SystemTime::duration::max();
}

(except without converting to duration)

Reply:
For context, here's some of the discussion we had about this last year, where we ultimately decided to keep this function (tho that might have been just due to me being stubborn, and should be reconsidered):
https://github.com/toddmgreer/envoy/pull/2#discussion_r330171353
https://github.com/toddmgreer/envoy/pull/2#discussion_r334191994

I recall not being able to use absl::Duration because it assumes a signed integer, and, at least at the time, we wanted to support unsigned values. Perhaps that's not a necessary constraint? It would be nice to consolidate how we parse integers and use just the one function.

Reply:
Thank you for the reminder--I'd completely forgotten that. I'm fine with either approach.

Reply:
From the thread linked, and Todd's comment above that we should keep code inside cache dir (different thread), my plan is to change the implementation to similar to `eatLeadingDuration`, but keep it inside `http_cache_utils`.
Is that an acceptable approach?